### PR TITLE
Mozilla Firefox Rapid Release 

### PR DESCRIPTION
**Pull request by open-sorcerer64**
Added Firefox, based on scripts by @chunky-milk 

here is tar gzip for testing
[Mozilla FIrefox RR.tar.gz](https://github.com/Botspot/pi-apps/files/6838274/Mozilla.FIrefox.RR.tar.gz)

### CODE REVIEW

**Comment by Itai-Nelken** [comment]
I have had stability ptobleme with this firefox.
Anyway the you can't use `purge-installed` in the uninstall script as you haven't used `pkg-install` in the install script.

I think its ok to use apt here, but I'm not sure. What do you say @Botspot?

**Comment by Tejas-Singh-py** [comment]
> I have had stability ptobleme with this firefox.

But, I my testing (I use it daily) it crashed only once or maybe twice, while logging in to the sync feature. After that it works like a charm.  

> Anyway the you can't use purge-installed in the uninstall script as you haven't used pkg-install in the install script.

Okay, makes sense. I have updated the PR and here is a `tarball` for testing

[Mozilla FIrefox RR.tar.gz](https://github.com/Botspot/pi-apps/files/6845103/Mozilla.FIrefox.RR.tar.gz)

Let's see what @Botspot thinks

**Comment by Itai-Nelken** [comment]
looks good. I'm not sure if you should use `pkg-install` to install firefox or just use APT (what your script does now). what do you say @Botspot?
@Tejas-Singh-py for now leave it as is.

**Comment by Tejas-Singh-py** [comment]
@Itai-Nelken, I didn't used `pkg-install` because it broke the install script.

File: apps/Mozilla FIrefox RR/install [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+function error {
+  echo -e "\\e[91m$1\\e[39m"
+  exit 1
+}
+
+# Add repository source to apt sources.list
+echo "deb http://ports.ubuntu.com/ubuntu-ports bionic-updates main" | sudo tee /etc/apt/sources.list.d/ubuntu.list || error "Failed to add repository source to apt sources.list!"
+
+# Add ubuntu keyring
+sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 3B4FE6ACC0B21F32 || error "Failed to add  ubuntu keyring!"
```
Comment (CleanMachine1) [inline apps/Mozilla FIrefox RR/install:14]: I am sure I could check easily later but not now, is this key timed or does it have an expire date?

File: data/categories/structure [hunk 2]
```diff
@@ -55,6 +55,7 @@ Minecraft Java|Games
 Minecraft Pi (Modded)|Games
 Minetest (MIneclone 2)|Games
 Mission Planner|Tools
+Mozilla FIrefox RR|Internet
```
Comment (CleanMachine1) [inline data/categories/structure:58]:
```suggestion
Mozilla Firefox RR|Internet
```

**Comment by CleanMachine1** [comment]
Why are all the files called FIrefox with a capital I

**Comment by CleanMachine1** [comment]
Like to just mention, this should be RR - Rolling release, however firefox devs went otherwise I guess.

Arch for the win :)

**Comment by Tejas-Singh-py** [comment]
> Why are all the files called FIrefox with a capital I

**TYPO**

**Comment by CleanMachine1** [comment]
All of the other files need to be changed, since they are all spelled with typos

**Comment by open-sorcerer64** [comment]
> All of the other files need to be changed, since they are all spelled with typos

How to fix this, new PR? I don't know much about Github PRs. If you find any guide please share it. @Itai-Nelken and @Botspot  any way to fix this except a new PR.

File: apps/Mozilla FIrefox RR/install (hunk 1 above)
Comment (open-sorcerer64) [reply apps/Mozilla FIrefox RR/install:14]: No idea. But AFAIK it isn't

**Comment by Itai-Nelken** [comment]
we'll just use the archive, not the pr.

**Comment by Itai-Nelken** [comment]
@open-sorcerer64 continued in #835

**Comment by CleanMachine1** [comment]
using `git mv` you could have done it very easily.

**Comment by open-sorcerer64** [comment]
> using `git mv` you could have done it very easily.

Using GitHub CLI? I am a noob in GitHub PRs, so....

**Comment by CleanMachine1** [comment]
No, git.

I dislike ghcli, git is superior once you learn how to use it just as the cherry-pick command 

Maybe try to learn git by owning a private repo and messing around

The project I work on (tldr) has a great collection of short little command examples for every git command

**Pull request closed without merging**